presto-connector-tests/src/main/java/com/facebook/presto/connector/unittest/MetadataTableTest.java

```diff
@@ -0,0 +1,238 @@
+@RequiredFeatures({CREATE_TABLE_AS, DROP_TABLE})
+public interface MetadataTableTest
+        extends BaseMetadataTest
+{
+    default void testRenameTable(SchemaTableName initial, SchemaTableName renamed)
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(
+                initial,
+                ImmutableList.of(
+                        new ColumnMetadata("bigint_column", BIGINT),
+                        new ColumnMetadata("double_column", DOUBLE)),
+                getTableProperties());
+
+        try (AutoCloseable cleanup = new AllCloser(withSchemas(session, schemaNamesOf(initial, renamed)))) {
+            run(ImmutableList.of(
+                    metadata -> {
+                        ConnectorOutputTableHandle handle = metadata.beginCreateTable(session, tableMetadata, Optional.empty());
+                        metadata.finishCreateTable(session, handle, ImmutableList.of());
+                    },
+                    metadata -> metadata.renameTable(session, metadata.getTableHandle(session, initial), renamed),
+                    metadata -> assertEquals(getOnlyElement(metadata.listTables(session, renamed.getSchemaName())), renamed),
+                    metadata -> metadata.dropTable(session, metadata.getTableHandle(session, renamed))));
+        }
+    }
+
+    @Test
+    @RequiredFeatures(RENAME_TABLE)
+    default void testRenameTableWithinSchema()
+            throws Exception
+    {
+        testRenameTable(
+                schemaTableName("initial"),
+                schemaTableName("renamed"));
+    }
+
+    @Test
+    @RequiredFeatures({CREATE_SCHEMA, DROP_SCHEMA})
+    default void testRenameTableAcrossSchema()
+            throws Exception
+    {
+        testRenameTable(
+                schemaTableName("initialSchema", "initialTable"),
+                schemaTableName("renamedSchema", "renamedTable"));
+    }
+
+    @Test
+    default void testListColumnsOneSchema()
+            throws Exception
+    {
+        SchemaTableName schemaTableName1 = schemaTableName("table1");
+        SchemaTableName schemaTableName2 = schemaTableName("table2");
+
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+
+        List<ColumnMetadata> columns = ImmutableList.of(
+                new ColumnMetadata("bigint_column", BIGINT),
+                new ColumnMetadata("double_column", DOUBLE));
+
+        List<ColumnMetadata> expectedColumns = withSystemColumns(columns);
+
+        ConnectorTableMetadata tableMetadata11 = new ConnectorTableMetadata(schemaTableName1, columns, getTableProperties());
+        ConnectorTableMetadata tableMetadata12 = new ConnectorTableMetadata(schemaTableName2, columns, getTableProperties());
+
+        try (AutoCloseable cleanup = withTables(session, tableMetadata11, tableMetadata12)) {
```

Comment:
Maybe we should keep `API` consistent for `withMetadata`.
 this:

```java
    default void withTables(ConnectorSession session, List<ConnectorTableMetadata> tables, Runnable runnable)
            throws Exception
    {
        try (AutoCloseable cleanup = withTables(session, tables.toArray(new ConnectorTableMetadata[0]))) {
            runnable.run();
        }
    }

    default void withTable(
            ConnectorSession session,
            ConnectorTableMetadata table,
            Runnable runnable)
            throws Exception
    {
        withTables(session, Arrays.asList(table), runnable);
    }

    default void withTables(
            ConnectorSession session,
            ConnectorTableMetadata table1,
            ConnectorTableMetadata table2,
            Runnable runnable)
            throws Exception
    {
        withTables(session, Arrays.asList(table1, table2), runnable);
    }

    default AllCloser withTables(ConnectorSession session, ConnectorTableMetadata... tables)
    {
        List<ConnectorTableMetadata> tableList = Arrays.asList(tables);
        return new AllCloser(Streams.concat(
                withSchemas(session, schemaNamesOf(tableList)).stream(),
                tableList.stream()
                        .map(table -> new Table(this, session, table)))
                .collect(toImmutableList()));
    }
```

Then this code snippet (with changes to `withMetadata` suggested above) would look like:
```java
        withTables(session, tableMetadata11, tableMetadata12, () ->
                withMetadata(
                        metadata -> assertEquals(metadata.listTableColumns(session, new SchemaTablePrefix()),
                                ImmutableMap.of(
                                        schemaTableName1, expectedColumns,
                                        schemaTableName2, expectedColumns)),
                        metadata -> assertEquals(metadata.listTableColumns(session, prefixOf(schemaTableName1)),
                                ImmutableMap.of(
                                        schemaTableName1, expectedColumns))));
```

Nicer?